app/controllers/katello/api/v2/host_collections_controller.rb

```diff
@@ -56,6 +56,7 @@ def index_relation
         query = HostCollection.readable
       end
       query = query.where(:name => params[:name]) if params[:name]
+      query.uniq unless query.is_a?(ActiveRecord::Associations::CollectionProxy)
```

Comment:
It looks like we could use `uniq!` to modify the query in place, while keeping it a CollectionProxy

```
irb(main):019:0> query = @host.host_collections
=> #<ActiveRecord::Associations::CollectionProxy []>

# keeps the class the same
irb(main):020:0> query.uniq!
=> #<ActiveRecord::Associations::CollectionProxy []>
irb(main):022:0> query.class
=> Katello::HostCollection::ActiveRecord_Associations_CollectionProxy

# changes to Array
irb(main):023:0> query = query.uniq
=> []
irb(main):024:0> query.class
=> Array

```